feat(command) adds a command to place the systemd service file in the right location

legendofmiracles opened the pull request:
This PR adds a command to generate a systemd service file and places it in the /etc/systemd/system, so that the user can easily enable a gameserver upon boot, without having to write his/her own service file.

### Why only systemd?

Because all the official supported distros use systemd and most distros in general as well.
If people request other init systems, i'd glady add support for those aswell.


* [x] New feature (change which adds functionality).

## Checklist
- [x] This pull request uses the `develop` branch as its base.
* [x] This pull request Subject follows the Conventional Commits standard.
* [x] This code follows the style guidelines of this project.
* [x] I have performed a self-review of my code.
* [x] I have checked that this code is commented where required.
* [x] I have provided a detailed with enough description of this PR.
* [x] I have checked If documentation needs updating.

## Documentation
EDIT: I opened a Pull request:
https://github.com/GameServerManagers/LinuxGSM-Docs/pull/54

## Showcase
(the last information text, prints "the game" instead of this)
![image](https://user-images.githubusercontent.com/30902201/101826334-c54ebd80-3b26-11eb-8224-3966b3696bc6.png)

dgibbs64 committed:
Merge branch 'release/v20.6.0' into develop

dgibbs64 committed:
Merge branch 'hotfix/v20.6.1' into develop

legendofmiracles committed:
feat: adds a command to place the systemd service file in the right location

dgibbs64 committed:
Merge branch 'hotfix/v20.6.2' into develop

dgibbs64 committed:
release v20.6.2

h3o66 commented:
Maybe you could squash/rebase the commits to one and do a commit title that matches the "Semantic Pull Request" standard.
More info can be find in the dev docs:
https://dev-docs.linuxgsm.com/getting-started/branching#pull-requests

alx365 commented:
Whoops, having some trouble with git ;_;

alx365 commented:
Is this better?

h3o66 commented:
> Is this better?

looks good to me

unknowntrojan committed:
fix(stop) stop will clear tmux text before entering stop command (#3142)

Tmux will ENTER before using stop command to clear any text entered in teh console

h3o66 committed:
fix(update): redo steamcmd update part for new behavoir (#3139)

Co-authored-by: Daniel Gibbs <me@danielgibbs.co.uk>

h3o66 committed:
fix(bmdm): add missing dependency ncurses libs (#3138)

Co-authored-by: Daniel Gibbs <me@danielgibbs.co.uk>

h3o66 committed:
fix(minecraft): fix java ram assignment (#3147)

dgibbs64 committed:
feat(config): add secrets config to store sensitive settings (#3080)

* feat(config): add secrets.cfg

* changed name of config

* updated wording

h3o66 committed:
feat(newserver): Colony Survival (#3075)

h3o66 commented on lgsm/functions/command_install_init.sh:
```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
```
please use the normal bash header `#!/bin/bash` instead and add the default header for the description of the file.

alx365 replied:
fixed

legendofmiracles committed:
change the shebang and adds the usual description of the file

h3o66 reviewed (commented):
ok for me

dgibbs64 committed:
feat(skeleton): New command skeleton (#3152)

* initial commit of skel

* getopt

* getopt

* getopt

* cpio

* adding messages

* exit

* message

* shellcheck

* truncate netstat for colserver

* added cpio dependency

* tabs

h3o66 committed:
feat(check-update): add new command check-update for steamcmd game servers (#3146)

legendofmiracles commented:
Should i maybe add a --dryrun flag, to print out the contents but not place them?

legendofmiracles committed:
replace the file with /usr/lib/ with /etc/

legendofmiracles commented:
I added documentation: https://github.com/GameServerManagers/LinuxGSM-Docs/pull/54

h3o66 committed:
fix(steamcmd): retries are not honored if install succeeds (#3155)

dgibbs64 committed:
feat: refactor check_config removing some game specific code (#3160)

h3o66 committed:
fix(gmod): update link for wiki page (#3165)

h3o66 committed:
fix(steamcmd): fix branch switch (#3168)

dgibbs64 committed:
Merge branch 'develop' into master

dgibbs64 commented on lgsm/functions/command_install_init.sh:
```diff
@@ -0,0 +1,43 @@
+fn_install_systemd
+if [ $? -eq 1 ]
+then
+	fn_print_error_nl "su exited with non 0 exit code.. something went wrong";
```
this should be a fail as it caused a fatal error

dgibbs64 commented on lgsm/functions/command_install_init.sh:
```diff
@@ -0,0 +1,43 @@
+else
+	fn_print_complete_nl "Placed the file in /etc/systemd/system/ as ${selfname}-lgsm.service"
+	fn_print_information_nl "run \`systemctl enable ${selfname}-lgsm.service\` (as root), to enable the game on boot"
+	fn_script_log_pass "sucessfullly installed the systemd service file"
```
spelling error "sucessfullly"

dgibbs64 commented on lgsm/functions/command_install_init.sh:
```diff
@@ -0,0 +1,43 @@
+	file_c=$(printf "$file_c" | sed s/'gamename'/$gamename/g | sed "s|path|$path|g" | sed s/'whoami'/$(whoami)/g)
+	fn_print_ok_nl "Generated the file contents"
+	fn_print_information_nl "Enter the password of root:"
+	su -c "echo \"$file_c\" > /etc/systemd/system/${selfname}-lgsm.service" || return 1 && return 0
```
I am having issues getting this to work. 

The only other time LinuxGSM tries to elevate to root is using `sudo` privileges for a user. However this is not always possible. I would suggest if the user is unable to elevate that a file is generated and the command is printed to screen so the user can do it manually from root.

dgibbs64 replied:
I am looking forward to getting this feature added. It is going to be a great addition to LinuxGSM :)

legendofmiracles replied:
So it should use sudo instead of su, and check if sudo failed in order to print the contents and instruct the user how/where to place those?

legendofmiracles replied:
It now uses sudo and prints the contents on error - with instructions where to put it.

legendofmiracles replied:
edit: i just saw that you wanted a command printed - will do

dgibbs64 reviewed (changes requested):
Great PR. I have provided feedback. Mainly noting that it is not always possible for a user to elevate to root and instead look at using sudo. If all else fails make it easy for the user to do it manually

dgibbs64 commented:
Also this will need to link to an issue and docs will need to be updated.

h3o66 committed:
feat(newserver): Vintage Story (#3156)

Co-authored-by: Daniel Gibbs <me@danielgibbs.co.uk>

h3o66 committed:
feat(newserver): SCP: Secret Laboratory (#3163)

h3o66 committed:
fix(rust): server config text change for rcon (#3175)

Co-authored-by: Daniel Gibbs <me@danielgibbs.co.uk>

h3o66 committed:
fix(steamcmd): remove unused parameter (#3170)

legendofmiracles commented:
@dgibbs64 this PR links to an issue: 
![image](https://user-images.githubusercontent.com/30902201/104278278-3d9afc80-54a0-11eb-86b5-458f5b511c84.png)
and i have also provided a update for the docs: https://github.com/GameServerManagers/LinuxGSM-Docs#54

legendofmiracles committed:
uses sudo instead of su and more helpful on errors

legendofmiracles committed:
Merge branch 'develop' into master

h3o66 committed:
fix(rust): wipe command does not wipe all sav files (#3183)

h3o66 committed:
feat(dev): add libc++.so.1 for dependency detection (#3181)

Co-authored-by: Daniel Gibbs <me@danielgibbs.co.uk>

h3o66 committed:
fix(pvr): add new dependency (#3180)

Co-authored-by: Daniel Gibbs <me@danielgibbs.co.uk>

h3o66 committed:
fix(inss): fix config to only use parameter (#3179)

h3o66 committed:
fix(steamcmd): fix update on start not starting the server (#3169)

Co-authored-by: Daniel Gibbs <me@danielgibbs.co.uk>

dgibbs64 committed:
Merge remote-tracking branch 'legendofmiracles/master' into systemd

h3o66 committed:
feat(steamcmd): set platformtype for steamcmd in the default config for the game server (#3167)

h3o66 committed:
fix(steamcmd): add dependency SDL 32bit for debian and ubuntu (#3186)

h3o66 committed:
fix(stop): change csgoserver stop mode (#3171)

Co-authored-by: Daniel Gibbs <me@danielgibbs.co.uk>

dgibbs64 committed:
Merge branch 'develop' into master
